cmd/clusterctl/pkg/client/init.go

```diff
@@ -104,17 +113,25 @@ type addToInstallerOptions struct {
 // addToInstaller adds the components to the install queue and checks that the actual provider type match the target group
 func (c *clusterctlClient) addToInstaller(options addToInstallerOptions, targetGroup clusterctlv1.ProviderType, providers ...string) error {
 	for _, provider := range providers {
+		// It is possible to opt-out from automatic installation of bootstrap/controlPlane providers using '-' as a provider name (NopProvider).
+		if provider == NopProvider {
+			if targetGroup == clusterctlv1.CoreProviderType {
```

Comment:
This will also allow people to use the `NopProvider` with the `InfrastructureProvider`, is that intentional/desired?

Reply:
I did not explicitly prevent uses to use `--infrastracture=-` because it can be useful to implement the n-infra multi-tenancy scenario described at https://master.cluster-api.sigs.k8s.io/clusterctl/commands/init.html#multi-tenancy.
For consistency, it is also allowed `--bootstrap=-` even if I don't see a use case for this now, while `--core=-` because it IMO does not really make sense